### ISSUE TITLE
Add a shim layer for libhdfs(3) API compatibility.

### ISSUE THREAD

**Issue by jamesclampffer** [haohui/libhdfspp#35]
It's useful, and in my case required, to add a pure C api on top of libhdfs++.  The best way to go about doing that is to implement what libhdfs and libhdfs3 already implement in the "hdfs.h" interface.  This also decouples internal changes from the outward facing API in preparation for renaming the InputStream class and any other refactoring that may happen. 

I've added a "chdfs.h" header file that implements a subset of these functions and I'll send a pull request momentarily.

Once this layer is merged into master I'll send another pull request for the tests I've built that use this API.

**Comment by jamesclampffer** [comment]
Pull request: https://github.com/haohui/libhdfspp/pull/36
